File: src/main/java/com/github/nsc/de/shake/parser/node/IfNode.java
package com.github.nsc.de.shake.parser.node;

import com.github.nsc.de.shake.lexer.characterinput.position.PositionMap;

public class IfNode extends Node {

    private final Tree body;
    private final Tree elseBody;
    private final ValuedNode condition;


    public IfNode(PositionMap map, Tree body, Tree elseBody, ValuedNode condition) {
        super(map);
        this.body = body;
        this.elseBody = elseBody;
        this.condition = condition;
    }


    public IfNode(PositionMap map, Tree body, ValuedNode condition) {
        this(map, body, null, condition);
    }

    public Tree getBody() {
        return body;
    }
    public Tree getElseBody() { return elseBody; }
    public ValuedNode getCondition() {
        return condition;
    }

    @Override
    public String toString() {
        return "IfNode{" +
                "body=" + body +
                ", condition=" + condition +
                '}';
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/IfNode.kt
package com.github.shakelang.shake.parser.node

import com.github.shakelang.shake.util.characterinput.position.PositionMap

class IfNode(map: PositionMap, val body: Tree, val elseBody: Tree?, val condition: ValuedNode) : Node(map) {

    constructor(map: PositionMap, body: Tree, condition: ValuedNode) : this(map, body, null, condition)

    override fun toJson(): Map<String, *> =
        mapOf(
            "name" to "IfNode",
            "condition" to condition.json,
            "body" to body.json
        )
}